### ISSUE TITLE
The test result is wrong

### ISSUE THREAD

**Issue by shliang0603** [albertpumarola/GANimation#22]
We use CK+ dataset, but when I use test dataset image to test ,the result doesn’t  expect,The following is the process of my train. Can you give me some suggestions?

1、Train GANimation
![image](https://user-images.githubusercontent.com/37186266/44559828-2946ef80-a77f-11e8-806f-5fc5887106a0.png)
![image](https://user-images.githubusercontent.com/37186266/44559832-2fd56700-a77f-11e8-90be-4cba1b5d0f4c.png)
![image](https://user-images.githubusercontent.com/37186266/44559835-3368ee00-a77f-11e8-8da1-91a86f998d8a.png)

2、The GPU working status
![image](https://user-images.githubusercontent.com/37186266/44559843-3ebc1980-a77f-11e8-9e9c-eecfbbcf4615.png)

3、The test result
![image](https://user-images.githubusercontent.com/37186266/44559849-47aceb00-a77f-11e8-8081-0fdc6f8deed2.png)
![image](https://user-images.githubusercontent.com/37186266/44559854-4bd90880-a77f-11e8-8611-4e5e52ad7086.png)

4、Train loss
![image](https://user-images.githubusercontent.com/37186266/44559856-5398ad00-a77f-11e8-9504-9d0ee044f10d.png)
![image](https://user-images.githubusercontent.com/37186266/44559858-572c3400-a77f-11e8-8357-9d4439f24d27.png)
![image](https://user-images.githubusercontent.com/37186266/44559860-5abfbb00-a77f-11e8-8e02-66f7150f918b.png)

5、The train image result
![image](https://user-images.githubusercontent.com/37186266/44559866-614e3280-a77f-11e8-9b41-e646351c0ca1.png)

**Comment by chenyvehtung** [comment]
It seems that your batch size is kind of too large, basically you'll expect to get results with better quality when training with a smaller batch size in the generative model. 
Or it can be possibly caused by some error in extracting the AU feature, which is a very important condition in this network architecture. 

I train this model with [CelebA](http://mmlab.ie.cuhk.edu.hk/projects/CelebA.html), some of my testing results are listed as follows, 
![190391_194900](https://user-images.githubusercontent.com/5866866/44567274-4b9f3400-a7a4-11e8-8d69-4cf9e59975d0.jpg)
![192092_194058](https://user-images.githubusercontent.com/5866866/44567275-4c37ca80-a7a4-11e8-8203-5bb55a216e79.jpg)
![194182_200262](https://user-images.githubusercontent.com/5866866/44567276-4c37ca80-a7a4-11e8-8f26-01b47206fbaf.jpg)
![199273_191153](https://user-images.githubusercontent.com/5866866/44567277-4cd06100-a7a4-11e8-8b81-cde00297170d.jpg)
Here, I try to change expression of the left most person to the right most one's. I interpolate it continuously by using the method mentioned in Section 6.2 of the original paper.

**But**, please note that the above shown results are **cherry picked** by me. There are also lots of failure results, even more than good results. Also I list some of them as follows,
![190143_200541](https://user-images.githubusercontent.com/5866866/44567863-e3058680-a7a6-11e8-8005-efa402a8ba24.jpg)
![190190_201477](https://user-images.githubusercontent.com/5866866/44567864-e3058680-a7a6-11e8-8d72-ce1b9ae00608.jpg)
![190304_201219](https://user-images.githubusercontent.com/5866866/44567865-e39e1d00-a7a6-11e8-9de2-1dd76bbf72a9.jpg)
![191839_194596](https://user-images.githubusercontent.com/5866866/44567866-e39e1d00-a7a6-11e8-955d-b573a7cff349.jpg)

I'm not sure it is because of my training data set or whatever. Or the quality of generated images are inherently unstable, just as lots of other GAN based model have shown.

**Comment by albertpumarola** [comment]
GANs are still unstable, but you can improve your results by training longer/more data/better data.

**Comment by shliang0603** [comment]
@chenyvehtung  Hi, Yuedong Chen. Can you package your data set , the processed AU, as well as the code and send me a copy ? I compare what went wrong.

**Comment by shliang0603** [comment]
@albertpumarola  Hi, albertpumarola. Can you send me a copy for your data set and processed AU ,I feel my processed AU is error.

**Comment by HrsPythonix** [comment]
@chenyvehtung What kind of image are you using in your training? I use the aligned one and find their size is fixed at 178 × 218，so in order to change them to 128*128, I add a centercrop and resize(128) into the transform in dataloader, but the training result is not good as expected. Could you tell me how to preprocess the dataset? Thanks.

**Comment by yangzhikai** [comment]
@chenyvehtung hello, the output images of openface is  *.bmp on my cumputer. Did you get the *.bmp images  too? That makes me "An ignorant face"........

**Comment by yangzhikai** [comment]
@shliang0603 How do you visualize it?There are no web link tutorials？thank

**Comment by joyyang1997** [comment]
@donydchen Hello, I use the same dataset as you,but the results seem to be bad. The output is almost the same as the input. I don't know what went wrong.
Have you modified the parameters of the network? If so, would you like to share your params? Thaks a lot!

**Comment by donydchen** [comment]
> @donydchen Hello, I use the same dataset as you,but the results seem to be bad. The output is almost the same as the input. I don't know what went wrong.
> Have you modified the parameters of the network? If so, would you like to share your params? Thaks a lot!

Sorry for the late reply. Actually I don't use the codes provided by this project. Instead, I replicate GANimation from scratch using the information illustrated in the original paper.

**Comment by Hshihua** [comment]
@shliang0603 Have you solved this issue? I encountered the same issue with the implementation by @donydchen.

**Comment by Hshihua** [comment]
@joyyang1997 Me too. Have you fixed that? I have tested the pretrained model provided by @donydchen, and it was pretty good. However, when I retrain the model with the provided codes, the results are bad. Any suggestions?
